src/utilstrencodings.cpp

```diff
@@ -229,7 +229,7 @@ vector<unsigned char> DecodeBase64(const char* p, bool* pfInvalid)
 string DecodeBase64(const string& str)
 {
     vector<unsigned char> vchRet = DecodeBase64(str.c_str());
-    return (vchRet.size() == 0) ? string() : string((const char*)&vchRet[0], vchRet.size());
+    return std::string((const char*)vchRet.data(), vchRet.size());
```

Comment:
I read that as: If the count is 0, it always points at least 0 elements,  thus "point at an array of at least count elements of CharT" is always satisfied. 

Alternatively: if count is 0 it would be wrong to access the pointer's address at all.